Fix project configuration to help contributors

## Environment

<!-- Run `react-native info` in your terminal and paste its contents here. -->
System:
    OS: Linux 5.3 Ubuntu 19.10 (Eoan Ermine)
    CPU: (8) x64 AMD Ryzen 5 2500U with Radeon Vega Mobile Gfx
    Memory: 713.05 MB / 14.62 GB
    Shell: 5.0.3 - /bin/bash
  Binaries:
    Node: 12.13.1 - ~/.nvm/versions/node/v12.13.1/bin/node
    npm: 6.13.7 - ~/.nvm/versions/node/v12.13.1/bin/npm
  npmPackages:
    react: 16.9.0 => 16.9.0 
    react-native: 0.61.2 => 0.61.2

## Description

<!--
  Describe your issue in detail. Include screenshots if needed. If this is a regression, let us know.
-->
The project configuration is messy and the scripts don't work out of the box.

## Reproducible Demo

<!--
  Let us know how to reproduce the issue. Include a code sample or share a project that reproduces the issue.
  Please follow the guidelines for providing a minimal example: https://stackoverflow.com/help/mcve.
-->

```
git clone git@github.com:react-native-community/react-native-slider.git
cd react-native-slider
npm i
npm run validate:eslint
```

And by the way, it seems that the current code base doesn't pass the validation:

```
yarn run v1.21.1
$ flow check src
Error ----------------------------------------------------------------------------------- example/SliderExample.js:15:20

Cannot resolve module `@react-native-community/slider`.

   15| import Slider from '@react-native-community/slider';
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^



Found 1 error
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

Exited with code exit status 2
```

I also got:

```
yarn run v1.21.1
$ jest src
 FAIL  src/js/__tests__/Slider.test.js
  ● Test suite failed to run

    Cannot find module '@babel/preset-flow' from '/home/circleci/project'
```

--- a/README.md
+++ b/README.md
@@ -265,6 +265,13 @@ Assigns a single image for the track. Only static images are supported. The cent
 
 ## Contributing
 
+This project uses [yarn workspaces](https://classic.yarnpkg.com/en/docs/workspaces/) to handle its internal dependencies.
+Make sure to use `yarn` to install dependencies
+
+```sh
+yarn install
+```
+
 While developing, you can run the [example app](/example/README.md) to test your changes.
 
 Make sure your code passes Flow, ESLint and the tests. Run the following to verify:
@@ -274,6 +281,7 @@ yarn validate:flow
 yarn validate:eslint
 yarn test:jest
 ```
+
 or 
 
 ```sh

